modules/client/src/internal/client/methods.ts

```diff
@@ -255,15 +256,32 @@ export class ClientMethods extends ClientCore implements IClientMethods {
     );
 
     if (tokenAddress && tokenAddress !== AddressZero) {
-      // If the target is an ERC20 token, ensure that the amount can be transferred
-      // Relay the yield steps to the caller as they are received
-      yield* this.updateAllowance(
-        {
-          amount: params.amount,
-          daoAddressOrEns: daoAddress,
-          tokenAddress,
-        },
+      // check current allowance
+      const tokenInstance = new Contract(
+        tokenAddress,
+        erc20ContractAbi,
+        signer,
+      );
+      const currentAllowance = await tokenInstance.allowance(
+        await signer.getAddress(),
+        daoAddress,
       );
+      yield {
+        key: DaoDepositSteps.CHECKED_ALLOWANCE,
+        allowance: currentAllowance.toBigInt(),
+      };
+      // if its lower than the needed, set it to the correct one
+      if (currentAllowance.lt(params.amount)) {
+        // If the target is an ERC20 token, ensure that the amount can be transferred
+        // Relay the yield steps to the caller as they are received
+        yield* this.setAllowance(
```

Comment:
If we "connect" the generator of `setAllowance` with our caller, then our caller will be getting the values of `deposit` plus the values of `setAllowance`.

However, the return type of the function is `AsyncGenerator<DaoDepositStepValue>` and doesn't include `SetAllowanceStepValue`

Reply:
`DaoDepositStepValue` does include `SetAllowanceStepValue`

```ts

export type DaoDepositStepValue =
  | SetAllowanceStepValue
  | { key: DaoDepositSteps.CHECKED_ALLOWANCE; allowance: bigint }
  | { key: DaoDepositSteps.DEPOSITING; txHash: string }
  | { key: DaoDepositSteps.DONE; amount: bigint };
```